packages/models/src/crypto-assets/crypto-asset.model.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  BtcCryptoAssetBalance,
+  CryptoAssetBalance,
+  StxCryptoAssetBalance,
+} from './crypto-asset-balance.model';
+import {
+  Brc20CryptoAssetInfo,
+  BtcCryptoAssetInfo,
+  InscriptionCryptoAssetInfo,
+  RuneCryptoAssetInfo,
+  Sip009CryptoAssetInfo,
+  Sip010CryptoAssetInfo,
+  StampCryptoAssetInfo,
+  Stx20CryptoAssetInfo,
+  StxCryptoAssetInfo,
+} from './crypto-asset-info.model';
+
+export type CryptoAssetType =
+  | 'btc'
+  | 'stx'
+  | 'brc-20'
+  | 'inscriptions'
+  | 'runes'
+  | 'sip-009'
+  | 'sip-010'
+  | 'src-20'
+  | 'stamps'
+  | 'stx-20';
+
+/**
+ * Cryptocurrencies - utility tokens
+ */
+export interface BtcCryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'btc';
+  readonly asset: BtcCryptoAssetBalance & BtcCryptoAssetInfo;
+}
+
+export interface StxCryptoAsset {
+  readonly blockchain: 'stacks';
+  readonly type: 'stx';
+  readonly asset: StxCryptoAssetBalance & StxCryptoAssetInfo;
+}
+
+/**
+ * Fungible and non-fungible tokens
+ */
+export interface Brc20CryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'brc-20';
+  readonly asset: CryptoAssetBalance & Brc20CryptoAssetInfo;
+}
+
+export interface RuneCryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'runes';
+  readonly asset: CryptoAssetBalance & RuneCryptoAssetInfo;
+}
+
+export interface InscriptionCryptoAsset {
+  readonly blockchain: 'bitcoin';
+  readonly type: 'inscriptions';
```

Comment:
As this represents a single item, shouldn't it be singular?

Reply:
The protocol name is typically plural, so wasn't sure here.